backend/src/zimfarm_backend/migrations/versions/12096716adfd_add_offliner_definitions.py

```diff
@@ -0,0 +1,167 @@
+"""add offliner definitions
+
+Revision ID: 12096716adfd
+Revises: 60c546babd23
+Create Date: 2025-09-25 00:45:04.365564
+
+"""
+
+import json
+from pathlib import Path
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy import update
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.orm import Session
+
+from zimfarm_backend.common.enums import Offliner
+from zimfarm_backend.common.schemas.offliners.builder import OfflinerFlagSchema
+from zimfarm_backend.db.models import RequestedTask, Schedule, Task
+from zimfarm_backend.db.offliner_definition import create_offliner_definition
+
+# revision identifiers, used by Alembic.
+revision = "12096716adfd"
+down_revision = "60c546babd23"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "offliner_definition",
+        sa.Column(
+            "id",
+            sa.Uuid(),
+            server_default=sa.text("uuid_generate_v4()"),
+            nullable=False,
+        ),
+        sa.Column("offliner", sa.String(), nullable=False),
+        sa.Column("definition", postgresql.JSON(astext_type=sa.Text()), nullable=False),
+        sa.Column("version", sa.String(), nullable=False),
+        sa.Column("created_at", sa.DateTime(), nullable=False),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk_offliner_definition")),
+        sa.UniqueConstraint(
+            "offliner", "version", name=op.f("uq_offliner_definition_offliner")
+        ),
+    )
+
+    # Set the foreign keys of the requested task, task and schedule to be nullable.
+    # After seeding, we alter it to False to reflect application requirements
+
+    op.add_column(
+        "requested_task", sa.Column("offliner_definition_id", sa.Uuid(), nullable=True)
+    )
+    op.create_foreign_key(
+        op.f("fk_requested_task_offliner_definition_id_offliner_definition"),
+        "requested_task",
+        "offliner_definition",
+        ["offliner_definition_id"],
+        ["id"],
+    )
+
+    op.add_column(
+        "schedule", sa.Column("offliner_definition_id", sa.Uuid(), nullable=True)
+    )
+    op.create_foreign_key(
+        op.f("fk_schedule_offliner_definition_id_offliner_definition"),
+        "schedule",
+        "offliner_definition",
+        ["offliner_definition_id"],
+        ["id"],
+    )
+
+    op.add_column("task", sa.Column("offliner_definition_id", sa.Uuid(), nullable=True))
+    op.create_foreign_key(
+        op.f("fk_task_offliner_definition_id_offliner_definition"),
+        "task",
+        "offliner_definition",
+        ["offliner_definition_id"],
+        ["id"],
+    )
+
+    data_dir = Path(__file__).parent.parent / "seed"
```

Comment:
Maybe rename folder from `seed` to `initial_offliner_definitions`

Reply:
Done